hazelcast/src/main/java/com/hazelcast/cache/impl/CacheProxySupport.java

```diff
@@ -92,7 +93,7 @@
     private static final int TIMEOUT = 10;
 
     protected final ILogger logger;
-    protected final CacheConfig<K, V> cacheConfig;
+    protected CacheConfig<K, V> cacheConfig;
```

Comment:
From which threads can this reassignment be called now? As I don't see safe publication of new values.

Reply:
I think you already changed this

Reply:
this is called when the tenant is undeployed and cache config is deserialized. It doesn't have to be propagated anywhere and cannot be inconsistent.

Reply:
The thread-safety semantics remain the same, I only removed duplication. 

>  It doesn't have to be propagated anywhere and cannot be inconsistent.
Can you elaborate? If that is so, why do we need to update it in the first place?

Reply:
To remove references to classes no longer available via class loader when application is I deployed from Payara

Reply:
I don't know how else to reserialize cache config when tenant is undeployed, to remove all user classes that are no longer available on the current instance, thus the title of original issue for this PR.
#10875

Reply:
Also I can't prove a negative

Reply:
The only other way I could see to do this is to recreate the whole proxy again with the new re-serialized cache config. I have no idea how to do that safely either.